src/components/Card/ProductCard/ProductCard.jsx

```diff
@@ -1,20 +1,52 @@
-import { Link } from "react-router-dom";
-import { useFilter } from "../../../contexts";
+import { FaRegHeart, FaHeart } from "react-icons/fa";
+import { Link, useNavigate } from "react-router-dom";
+import { useFilter, useCart, useAuth } from "../../../contexts";
 import styles from "./ProductCard.module.css";
+import { Button } from "../../index";
 export const ProductCard = (props) => {
-  const { img, name, price, description } = props;
+  const { _id, img, name, price, description, isLiked } = props;
   const { setShowProduct } = useFilter();
+  const { loggedIn } = useAuth();
+  const { cartDispatch, postToCart, cartState } = useCart();
+  const { cart } = cartState;
+  const token = localStorage.getItem("UserToken");
+  const navigate = useNavigate();
+  const addToCart = async (product) => {
+    try {
+      const newCart = await postToCart(product, token);
+      if (newCart) {
+        cartDispatch({ type: "ADD_TO_CART", payload: newCart });
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
   return (
-    <Link to="/productDetails">
-      <div className={styles.card} onClick={() => setShowProduct(props)}>
-        <img src={img} alt={name} className="card-img" />
-        <div className="card-close-icon"></div>
+    <div className={styles.card} onClick={() => setShowProduct(props)}>
+      <img src={img} alt={name} className="card-img" />
+      <div className="card-close-icon">
+        {isLiked ? <FaHeart color={"red"} /> : <FaRegHeart />}
+      </div>
+      <Link to="/productDetails">
         <div className="card-header">
           <h3 className="headline-3">{name}</h3>
           <p className="small-text-2">Rs.{price}</p>
```

Comment:
You can use the ruppee symbol using `&#8377;`

Reply:
Okay Will do that